### PR TITLE
sim: Add a CLI option to override vpnkit socket

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

File: arch/sim/src/sim/up_head.c [hunk 1]
```diff
@@ -77,6 +78,8 @@ static char g_logbuffer[4096];
 
 int main(int argc, char **argv, char **envp)
 {
+  process_cli_options(argc, argv);
```
Comment (xiaoxiang781216) [inline arch/sim/src/sim/up_head.c]: How about we just save argc and argv in some global variable? Then each component could invoke getopt to parse the argument who is interesting. The advantage is that:
1.Don't need expose the component internal detail to the world
2.Don't need modify the common code to add the new option

Comment (yamt) [reply arch/sim/src/sim/up_head.c]: @xiaoxiang781216 with your suggestion, how can component-A skip options for component-B?

Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_head.c]: Just ignore the option he doesn't understand.

File: boards/sim/sim/sim/NETWORK-VPNKIT.txt [hunk 2]
```diff
@@ -34,6 +35,3 @@ You can use it as the following:
 
     % vpnkit --ethernet /tmp/vpnkit-nuttx &
     % ./nuttx
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/NETWORK-VPNKIT.txt]: How about add --vpnkit-sock description here?

File: arch/sim/src/sim/up_head.c (hunk 1 above)
Comment (yamt) [reply arch/sim/src/sim/up_head.c]: getopt doesn't work that way. or are you suggesting to stop using getopt?

File: boards/sim/sim/sim/NETWORK-VPNKIT.txt (hunk 2 above)
Comment (yamt) [reply boards/sim/sim/sim/NETWORK-VPNKIT.txt]: done

File: arch/sim/src/sim/up_head.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_head.c]: > getopt doesn't work that way. or are you suggesting to stop using getopt?

Why we can't use getopt like this:
```
  while ((option = getopt(argc, argv, "f")) != ERROR)
    {
      if (option == 'f')
        {
           ...
        }
   }
```

Comment (yamt) [reply arch/sim/src/sim/up_head.c]: @xiaoxiang781216 
how would it skip an unknown option, say, -g?
note that -g might or might not take an argument.

Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_head.c]: > @xiaoxiang781216
> how would it skip an unknown option, say, -g?
> note that -g might or might not take an argument.

Yes, getopt can't handle the option with argument. But I still think the code struct is better even we have to parse the argument directly, do you think so?

Comment (yamt) [reply arch/sim/src/sim/up_head.c]: it depends.
if we invent a "generic" framework, how many components would use it?
if it's just a few or several, i guess #ifdef is good enough.

Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_head.c]: BTW, is it enough that we just provide Kconfig like tap bridge?

Comment (yamt) [reply arch/sim/src/sim/up_head.c]: maybe. a command line option is more convenient for me though.

may i move the command line option part to a separate PR? i haven't expected it was this controversial.

Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_head.c]: > maybe. a command line option is more convenient for me though.
> 
> may i move the command line option part to a separate PR? i haven't expected it was this controversial.

Sure, thanks for your support.
BTW, how about we use environment variable which is more easier than command line for extension.

Comment (yamt) [reply arch/sim/src/sim/up_head.c]: i agree environment variables don't have modularity issues. i feel it's a bit backdoor-ish though.

**Comment by yamt** [comment]
i'll create separate PRs.

**Pull request closed without merging**